### PR TITLE
update check-interface-throughput-snmp rule

### PR DESCRIPTION
**Pull request by bmadhavi74**

### CODE REVIEW

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 1]
```diff
@@ -1,11 +1,282 @@
...
+                * the threshold ($threshold %).
+                */
+                term is-input-bandwidth-abnormal {
+                    when {
+                        greater-than-or-equal-to "$in-bw-util" "$threshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "$interface-name input throughput utilization ($in-bw-util %) crossed threshold value ($threshold %). Mbps:$in-mbps Bps:$input-bytes bps";
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:195]: it (Bps:$input-bytes bps) is bytes not bps hence same. cane be removed

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 2]
```diff
@@ -1,11 +1,282 @@
...
+                    }
+                }
+                /*
+                * Defaults color to green.
+                */
+                term is-input-bandwidth-normal {
+                    then {
+                        status {
+                            color green;
+                            message "$interface-name input throughput utilization ($in-bw-util %) is normal. Mbps:$in-mbps Bps:$input-bytes bps ";
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:206]: it (Bps:$input-bytes bps) is bytes not bps hence same. cane be removed

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 3]
```diff
@@ -1,11 +1,282 @@
...
+                * than the threshold ($threshold %).
+                */
+                term is-output-bandwidth-abnormal {
+                    when {
+                        greater-than-or-equal-to "$out-bw-util" "$threshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "$interface-name output throughput utilization($out-bw-util%) crossed threshold value ($threshold %). Mbps:$out-mbps Bps:$output-bytes bps";
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:227]: it (Bps:$output-bytes bps) is bytes not bps hence same. cane be removed

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 4]
```diff
@@ -1,11 +1,282 @@
...
+                    }
+                }
+                /*
+                * Defaults color to green.
+                */
+                term is-output-bandwidth-normal {
+                    then {
+                        status {
+                            color green;
+                            message "$interface-name output throughput utilization ($out-bw-util %) is normal. Mbps:$out-mbps Bps:$output-bytes bps";
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:238]: it (Bps:$output-bytes bps) is bytes not bps hence same. cane be removed

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 5]
```diff
@@ -1,11 +1,282 @@
...
+                    }
+                }
+            }
+            /*
+            * Variables with default values. Default values can be changed
+            * while deploying playbook instance.
+            */
+            variable interface-name-variable {
+                value .*;
+                description "Interfaces to be monitored";
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:249]: s/Interfaces to be monitored/Input interfaces to be monitored in regex/g

File: community_supplied/Interface/snmp-interface-throughput.rule [hunk 6]
```diff
@@ -1,11 +1,282 @@
...
+            }
+            /*
+            * Sensor configuration to get data from network devices.
+            */
+            sensor input-output-bytes {
+                synopsis "Interface SNMP sensor definition";
+                description "Interfaces SNMP sensor to data from network device";
+                snmp {
+                    table IF-MIB::ifXTable;
+                    frequency 60s;
```
Comment (gvarao) [inline community_supplied/Interface/snmp-interface-throughput.rule:47]: Increase frequency from 60s to 180s

Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Addressed and Increased frequency from 60s to 180s

File: community_supplied/Interface/snmp-interface-throughput.rule (hunk 5 above)
Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Changed "Interfaces to be monitored" to "Enter interface name in regex i.e. ge-.*"

File: community_supplied/Interface/snmp-interface-throughput.rule (hunk 4 above)
Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Addressed and changed the message from "Bps:$output-bytes bps" to "Output-bytes:$output-bytes"

File: community_supplied/Interface/snmp-interface-throughput.rule (hunk 3 above)
Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Addressed and changed the message from "Bps:$output-bytes bps" to "Output-bytes:$output-bytes"

File: community_supplied/Interface/snmp-interface-throughput.rule (hunk 2 above)
Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Addressed and changed the message from "Bps:$input-bytes bps" to "Input-bytes:$input-bytes"

File: community_supplied/Interface/snmp-interface-throughput.rule (hunk 1 above)
Comment (bmadhavi74) [reply community_supplied/Interface/snmp-interface-throughput.rule]: Addressed and changed the message from "Bps:$input-bytes bps" to "Input-bytes:$input-bytes"

**Comment by bmadhavi74** [comment]
- Added Dynamic threshold fields for both input and output bw utilization (dt-in-bw-util) and (dt-out-bw-util) respectively.

- Added yellow(medium) trigger handling the dynamic thresholds.

**Pull request merged**